pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -6,8 +6,21 @@ import { getLogger } from '../logger';
 import { type Stdio } from '../stdio';
 import { InterpreterClient } from '../interpreter_client';
 
+type AppConfigStyle = AppConfig & { terminal?: boolean | 'auto'; docked?: boolean | 'docked' };
+
 const logger = getLogger('py-terminal');
 
+const validate = (name: string, dflt: string, value?: boolean | string) => {
```

Comment:
Do you think `dflt` should be renamed something else? I had to look at it twice 😄

Reply:
yes, I vote for calling it `default_`

Reply:
@WebReflection this is a genuine question which shows all my ignorance of this stuff: why are you using an anonymous function instead of using the `function` statement?
Is there any semantics difference? Any advantage/disadvantage for using either one?

Or it's just a style preference?
(Personally, I find `function` much more readable but I'm happy to learn to use the proper idioms and conventions which are used by the JS community)

Reply:
Ooops, I've answered in here ... I can copy over that answer:
https://github.com/pyscript/pyscript/pull/1284/files#r1136927512

Reply:
Let me do that ...

> why are you using an anonymous function instead of using the function statement?

the correct terminology is *arrow function* because `function () {}` or `Function('')` are also anonymous (actually explicitly anonymous in the `Function` case).

Arrow functions have the following features:

  * these are not constructable so these are more explicit in their intent (helpers, utilities, not classes or functions with prototype): `new (() => {})` throws an error and no `prototype` exists at all
  * these do not pass through the context and arguments creation dance: both `this` and `arguments` don't exist, so these are considered lightweight functions
  * because of the previous point, if used within methods for any purpose, there's never the need to `.bind(this)`
  * minifiers produce smaller layout (as plain text)
  * last, but not least, these are shorter/faster to type, similarly like lambdas in python are used, whenever convenient

Function declarations are also hoisted, which is a whole shenanigan a part, while arrows are always assigned, example:

```js
// this works fine in JS
wut();

function wut() {
  console.log('wait ...');
}
```

```js
// this would throw
wut();

const wut = () => {
  console.log('wait ...');
};
```

So, for the same reason nobody uses `var` anymore, using functions when neither the `this` context nor `arguments` is needed, is consider legacy/anti-pattern and I hope the reasons I gave you make sense.

I've seen another file using *arrows* instead o function, but I would've asked soon otherwise why `function` declarations are used everywhere else instead ... it's not an issue in general, but there are extra footguns, variables shadowed, hoisting, unnecessary operations behind the scene, that if can be avoided, why not?

That's my thinking or feeling around this topic.

Reply:
I'll never have opinions around Python as I'm no expert and every time it looks just beautiful to my eyes but it's worth adding a tiny detail to my explanation that should also definitively answer your initial question:

> why are you using an anonymous function

Arrow functions, as well as shorthand methods declarations such as `{method() {}}` *are always named* just like functions declarations: the name is inferred:

```js
const callback = () => { throw new Error('oh no!'); };
callback();

Uncaught Error: oh no!
    callback debugger eval code:1 <-- there you go!
    <anonymous> debugger eval code:1
```

The only case where an arrow is not named is with [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) but that's true for regular `function` expressions or declarations too.

I hope this completes concerns and reasons!

Reply:
thank you for the deep explanation!

> I hope this completes concerns and reasons!

sure, I didn't have any real *concern*, I just wanted to learn more :)